datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -50,9 +53,8 @@ pub trait FileOpener: Unpin {
     fn open(
         &self,
         store: Arc<dyn ObjectStore>,
-        file: ObjectMeta,
-        range: Option<FileRange>,
-    ) -> FileOpenFuture;
+        file_meta: FileMeta,
```

Comment:
Why a new struct and not just `PartitionedFile`

Reply:
because FileStream did **move** ObjectMeta into `FileOpener`, where `partition_values` field of `PartitionedFile` is kept as part of `FileStreamState`. Alternatively we could pass `PartitionedFile` reference to `FileOpener` or move `PartitionedFile` into `FileOpener` and then return `partition_values` from `FileOpener`. How would you like to approach that ?

Reply:
I mean you could just clone `partition_values` but fair enough